main_test.go

```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"testing"
+
+	"github.com/elastic/hey-apm/conv"
+	"github.com/elastic/hey-apm/strcoll"
+	"github.com/stretchr/testify/assert"
+)
+
+// All JSON fields in Input are required, with zero values being meaningful.
+func TestDefaultInput(t *testing.T) {
+	os.Args[1] = "-bench"
```

Comment:
@graphaelli do you know other way of doing this? `flag.Set` is not doing what I expected, or I am not doing it right.

Reply:
I think you'd allow creation of a [`FlagSet`](https://golang.org/pkg/flag/#NewFlagSet) from the tests and pass that in to `parseFlags()` in `main`, setting flag on that instead of via the module level functions.  https://golang.org/src/flag/flag_test.go might serve for inspiration